Core/GDCore/Extensions/Builtin/MouseExtension.cpp

```diff
@@ -361,13 +403,50 @@ void GD_CORE_API BuiltinExtensionsImplementer::ImplementsMouseExtension(
           _("Multitouch"),
           "res/conditions/touch.png")
       .AddCodeOnlyParameter("currentScene", "")
+      .AddParameter("expression", _("Touch index"))
+      .SetHidden();
+
+  extension
+      .AddCondition(
+          "HasAnyTouchOrMouseStarted",
+          _("A new touch has started"),
+          _("Check if a touch has just started or the mouse left button has "
+            "been pressed on this frame. The touch identifiers can be "
+            "accessed using StartedTouchOrMouseId() and StartedTouchOrMouseCount()."),
+          _("A new touch has started"),
+          _("Multitouch"),
+          "res/conditions/touch24.png",
+          "res/conditions/touch.png")
+      .AddCodeOnlyParameter("currentScene", "");
+
+  extension
+      .AddExpression(
+          "StartedTouchOrMouseCount",
+          _("Started touch count"),
+          _("The number of touches (including the mouse) that have just "
+            "started on this frame. The touch identifiers can be "
+            "accessed using StartedTouchOrMouseCount()."),
+          _("Multitouch"),
+          "res/conditions/touch.png")
+      .AddCodeOnlyParameter("currentScene", "");
+
+  extension
+      .AddExpression(
+          "StartedTouchOrMouseId",
+          _("Started touch identifier"),
+          _("The identifier of the touch or mouse that has just started on "
+            "this frame. The touch number of touches can be "
+            "accessed using StartedTouchOrMouseCount()."),
+          _("Multitouch"),
+          "res/conditions/touch.png")
+      .AddCodeOnlyParameter("currentScene", "")
       .AddParameter("expression", _("Touch index"));
 
   extension
       .AddCondition(
           "HasTouchEnded",
           _("A touch has ended"),
-          _("Check if a touch has ended."),
+          _("Check if a touch has ended or a mouse left button has been released."),
```

Comment:
This is a change to an existing condition.
Should this be SetHidden and a new condition `HasTouchIrMouseEnded` - like the previous?

Reply:
The touch id for the mouse is given at the start, there is no reason to check the mouse id at the end as the legacy expressions wont give it.